### ISSUE TITLE
`ionic g` will not make folder?

### ISSUE THREAD

**Issue by rdlabo** [ionic-team/ionic-cli#3475]
https://beta.ionicframework.com/docs/building/migration is written like structure.

```
src/
├── app/
　   ├── components/
　   ├── services/
　   └── pages/
```

But starter template and `ionic g` don't support this structure. I have not chased for a long time, what has become of this problem?

If Ionic team is waiting for implementation, I would like to challenge this. or If it keeps going, I would like to make option `ionic g -d`(--directory).

I believe that Ionic v3 structure supported many beginner's learning, this structure is great. so 
I also want to make it standard specification in v4.

thanks.

**Comment by dwieeb** [comment]
Haha, I literally just made some updates to those docs: https://github.com/ionic-team/ionic-docs/commit/a9cf5f3f564c37d13d70a821858366fbf112d269

Basically, we don't want to **force** apps into a structure, we want Angular to lead the way in recommended project structure. See original issue for more info: https://github.com/ionic-team/ionic-cli/pull/3392

**Comment by rdlabo** [comment]
Haha, Oh...It's good timing.

Sure. I understand. But distinguish between pages or components is too difficult for beginner. 
So I challenge -d option.(not force) Please review it if I make a draft.
thanks.

**Comment by dwieeb** [comment]
I agree it is difficult for a beginner. Hopefully the file names will be enough. We add the `.page.ts` suffix:

```
❯ l src/app/home
total 40
drwxr-xr-x   7 dan  staff   224B Jul 25 17:09 .
drwxr-xr-x  22 dan  staff   704B Jul 30 16:49 ..
-rw-r--r--   1 dan  staff   477B Jul 25 15:54 home.module.ts
-rw-r--r--   1 dan  staff   275B Jul 25 15:54 home.page.html
-rw-r--r--   1 dan  staff     1B Jul 25 15:54 home.page.scss
-rw-r--r--   1 dan  staff   669B Jul 25 15:54 home.page.spec.ts
-rw-r--r--   1 dan  staff   179B Jul 25 15:54 home.page.ts
```

vs component:

```
❯ l src/app/blah
total 24
drwxrwxr-x   6 dan  staff   192B Jul 30 16:35 .
drwxr-xr-x  22 dan  staff   704B Jul 30 16:49 ..
-rw-rw-r--   1 dan  staff    24B Jul 30 16:35 blah.component.html
-rw-rw-r--   1 dan  staff     0B Jul 30 16:35 blah.component.scss
-rw-rw-r--   1 dan  staff   621B Jul 30 16:35 blah.component.spec.ts
-rw-rw-r--   1 dan  staff   266B Jul 30 16:35 blah.component.ts
```

**Comment by rdlabo** [comment]
This is nice idea and enough for me.
But I think that still missing for begenners.

ex)
```
app-routing.module.ts
app.component.html
app.component.spec.ts
app.component.ts
app.module.ts
area/ <= this is components
auth.service.spec.ts
auth.service.ts
home/ <= this is page
login/ <= this is page
```

My friend's web designer loves Ionic. His skill is used jQuery a little（used plugin, can copy. but write script hardly）. But be abled to release apps used by Ionic.

Beginners will soon get lost file and folder. so I want to support beginners through folder structure.

**Comment by dwieeb** [comment]
@rdlabo Why did you close your PR?

**Comment by rdlabo** [comment]
@dwieeb Sorry. I failed to delete branch. Restore now!
